client/web/src/repo/settings/RepoSettingsMirrorPage.tsx

```diff
@@ -207,6 +210,70 @@ const CheckMirrorRepositoryConnectionActionContainer: React.FunctionComponent<
     )
 }
 
+// Add interface for props then create component
+interface CorruptionLogProps {
+    repo: SettingsAreaRepositoryFields
+    history: H.History
+}
+
+const CorruptionLogsContainer: React.FunctionComponent<CorruptionLogProps> = props => {
+    const health = props.repo.mirrorInfo.isCorrupted ? (
+        <Alert className={classNames('mb-0', styles.alert)} variant="danger">
+            The repository is corrupt, check the log entries below for more info and consider recloning.
+        </Alert>
+    ) : (
+        <Alert className={classNames('mb-0', styles.alert)} variant="success">
+            The repository is currently not corrupt
+        </Alert>
+    )
+
+    const logEvents: any[] = props.repo.mirrorInfo.corruptionLogs.map((log, index) => (
+        <li key={`${props.repo.name}#${index}`} className="list-group-item px-2 py-1">
```

Comment:
We can avoid the usage of `any` here by relying on the Typescript type inference. It correctly infers the type of `logEvents` as `JSX.Element[]`. And to avoid violating the [react/no-array-index-key](https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules/no-array-index-key.md) ESLint rule, we can use `log.timestamp` instead.

```suggestion
    const logEvents = props.repo.mirrorInfo.corruptionLogs.map((log, index) => (
        <li key={`${props.repo.name}#${log.timestamp}`} className="list-group-item px-2 py-1">
```